api/mutualfunds/tasks.py

```diff
@@ -45,17 +45,18 @@ def fetch_nav(self, scheme_ids=None, update_portfolio_kwargs=None):
             else:
                 from_date = datetime.date(1970, 1, 1)
                 logger.info("Fetching NAV for %s from beginning", scheme.name)
-            mfapi_url = f"https://api.mfapi.in/mf/{scheme.amfi_code}"
-            response = requests.get(mfapi_url, timeout=60)
-            data = response.json()
-            for item in reversed(data["data"]):
-                date = date_parse(item["date"], dayfirst=True).date()
-                if date <= from_date:
-                    continue
-                NAVHistory.objects.get_or_create(
-                    scheme_id=scheme.id, date=date, defaults={"nav": item["nav"]}
-                )
-            time.sleep(2)
+            if from_date < date.today() - timedelta(days=1) :
```

Comment:
@coderabbiai , please help me to write Celaary worker test

Reply:
@coderabbitai, how to fix the issue with this line, All I want to do is if from_date is less than today and if today is either saturday or sunday then set to friday only then proceed to next steps.

Reply:
@coderabbitai, this is throwing below error

```
 UnboundLocalError: cannot access local variable 'date' where it is not associated with a value
```

Reply:
@coderabbitai, I have added the import as below

```
from datetime import date, datetime, timedelta
```
but still getting the same error and in IDE i can see below error

```
"date" is not accessed
```

Reply:
@coderabbitai, above declaration is already defined, in spite of then receiving the error, Can you please read the whole file again and suggest the fix.